unison-cli/src/Unison/CommandLine/InputPatterns.hs

```diff
@@ -1243,25 +1244,27 @@ parseUri label input =
   let printError err = P.lines [P.string "I couldn't parse the repository address given above.", prettyPrintParseError input err]
    in first printError (P.parse UriParser.repoPath label (Text.pack input))
 
-prettyPrintParseError :: String -> P.ParseError Char Void -> P.Pretty P.ColorText
-prettyPrintParseError input = \case
-  P.TrivialError sp ue ee ->
-    P.lines
-      [ printLocation sp,
-        P.newline,
-        printTrivial ue ee
-      ]
-  P.FancyError sp ee ->
-    let errors = foldMap (P.string . mappend "\n" . P.showErrorComponent) ee
-     in P.lines
-          [ printLocation sp,
-            errors
-          ]
+prettyPrintParseError :: String -> P.ParseErrorBundle Text Void -> P.Pretty P.ColorText
+prettyPrintParseError input errBundle =
+  let (firstError, sp) = NE.head . fst $ P.attachSourcePos P.errorOffset (P.bundleErrors errBundle) (P.bundlePosState errBundle)
+   in case firstError of
+        P.TrivialError _errorOffset ue ee ->
+          P.lines
+            [ printLocation sp,
+              P.newline,
+              printTrivial ue ee
+            ]
+        P.FancyError _errorOffset ee ->
+          let errors = foldMap (P.string . mappend "\n" . showErrorFancy) ee
+           in P.lines
+                [ printLocation sp,
+                  errors
+                ]
```

Comment:
We should remember that we have the option to try printing multiple errors, and maybe try previewing what that might look like?